python/plugins/processing/algs/gdal/GdalUtils.py

```diff
@@ -609,3 +611,66 @@ def tr(cls, string, context=""):
         if context == "":
             context = cls.__name__
         return QCoreApplication.translate(context, string)
+
+    @staticmethod
+    def gdal_wms_xml_description_file(
+        uri: str,
+        version: str,
+        extent: QgsRectangle,
+        width: int,
+        height: int,
+        out_path: str,
+    ) -> tuple[bool, str]:
+        """
+        Creates an XML description file to access a WMS via GDAL.
+
+        Returns True if the file was created, or False if it was not.
+        An additional returned value provides callers with more details on an eventual write error.
+        """
+        # Prepare data
+        source = QgsProviderRegistry.instance().decodeUri("wms", uri)
+        base_url = source.get("url", "")
+        crs = source.get("crs", "EPSG:4326")
```

Comment:
I don't think 4326 should be hardcoded. What does GDAL default to if this isn't specified?

Reply:
Defaults to 4326, but I agree, the CRS should always be provided by the caller of our method.